datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1209,7 +1209,14 @@ impl SMJStream {
                     ) {
                         // The reverse of the selection mask. For the rows not pass join filter above,
                         // we need to join them (left or right) with null rows for outer joins.
-                        let not_mask = compute::not(mask)?;
+                        let not_mask = if mask.null_count() > 0 {
```

Comment:
I ran the added test by #9080 again in a new laptop and found this bug. I'm not sure why previously the test passed locally and in CI in #9080. 🤔

Reply:
I re-checked the results in `sort_merge_join.slt` and it should be correct (as it is same as `join.slt` which is produced by hash join operator).